### ISSUE TITLE
Can't use the new Assets loader with pixi.js

### ISSUE THREAD

**Issue by liana-p** [pixijs/pixijs#8740]
### Expected Behavior
The [docs](https://pixijs.download/dev/docs/PIXI.Assets.html) show how to use the assets loader:

```ts
import { Assets } from 'pixi.js';
```

### Current Behavior
This doesn't work:

<img width="607" alt="image" src="https://user-images.githubusercontent.com/32243840/196050293-10addb83-c3e3-4b00-b14b-efe51fd41b28.png">

At runtime: `Uncaught SyntaxError: The requested module '/node_modules/.vite/deps/pixi__js.js?v=25d6e02b' does not provide an export named 'Assets' (at plugin.ts:8:23)`

This is on pixi 6.5.6 which as far as I know is the default.

### Other attempt with `@pixi/assets`

Seeing that, I digged around and found this repo which seems to be in the process of converting to a monorepo (though I couldn't find any docs about that). I then found the `@pixi/assets` package and added it to my project.

This worked up to a point: When creating animated sprites using textures from the `Assets` package I get a strange error:

<img width="639" alt="image" src="https://user-images.githubusercontent.com/32243840/196050449-cf347a1c-b042-406f-bf37-88cd5654eae3.png">

This happens because `this._texture` in `AnimatedSprite` is undefined, which in turns is caused by the following:

<img width="636" alt="image" src="https://user-images.githubusercontent.com/32243840/196050490-34791fb0-3824-4ad3-9554-efce4995f4c7.png">

<img width="227" alt="image" src="https://user-images.githubusercontent.com/32243840/196050504-2f7bf010-fd58-4833-a1b8-d2b45eb7a271.png">

The textures coming from the `Assets` loader are not instances of the same base texture class as the one being checked inside `AnimatedSprite`.

This leads me to believe the reason this doesn't work is that using the `@pixi/assets` package on top of the normal `pixi.js` package probably ends up with two versions of the same files being imported, and thus two versions of the base texture class, maybe. Though when I looked at the monorepo, the `Assets` package seems to have no dependencies, so I don't know.

### Possible Solution

I don't know, I just expected to be able to import `Assets` from `pixi.js` as said in the docs.

I wouldn't mind using the monorepo package otherwise, but this is undocumented and I don't know if it's intended to be used.

### Steps to Reproduce

* Try to import `Assets` from `pixi.js`
* It doesn't exist

### Environment

- **`pixi.js` version**: 6.5.6
- **Browser & Version**: Chrome Version 106.0.5249.119 (Official Build) (arm64)
- **OS & Version**: MacOS
- **Note**: I am running this using [Vite](https://vitejs.dev). Though I don't think that should matter as this is standard import behaviour.

**Comment by SuperSodaSea** [comment]
`import { Assets } from 'pixi.js'` is for PixiJS v7, for v6 you still need `import { Assets } from '@pixi/assets'`.

**Comment by liana-p** [comment]
> `import { Assets } from 'pixi.js'` is for PixiJS v7, for v6 you still need `import { Assets } from '@pixi/assets'`.

Ok but I also tried using the `@pixi/assets` package as mentioned above, and it ends up in the weirdness described in my issue where textures are an instance of a different base texture class and the code ends up setting them to undefined.

I've now switched to trying to use the old loader and then getting my frames using `PIXI.Loader.shared.resources['my-sprite.json'].data`, except with this one there are no textures populated there and just the id of the sprites

So either way doesn't work and I've run out of ideas on how to create an animated spritesheet

**Comment by SuperSodaSea** [comment]
I tried this and reproduced your problem:

```
npm uninstall pixi.js @pixi/assets
npm install pixi.js@6.5.5 @pixi/assets@6.5.6
```

![image](https://user-images.githubusercontent.com/8724868/196083362-489cd320-9fb2-411c-b4c5-5ea760582529.png)

Seems that there are multiple versions of PixiJS in your `node_modules`. Can you try [this solution](https://github.com/pixijs/pixijs/wiki/Upgrading-PixiJS) in Wiki and see if it works?

```
npm uninstall pixi.js @pixi/assets
npm install pixi.js @pixi/assets
```

(If any other packages using PixiJS, also reinstall them.)

**Comment by Makio64** [comment]
@liana-p After multiple similar issues while updating pixi I arrived to the solution : with yarn everytime before i update pixi.js i just delete the yarn.lock and then run yarn install, no more incident since then.

**Comment by bigtimebuddy** [comment]
Unfortunately this is a long-stand annoying issue with v6 and how different package manager resolve peer dependencies. It has been addressed in v7 and upgrading should be very straight-forward.

**Comment by liana-p** [comment]
> Unfortunately this is a long-stand annoying issue with v6 and how different package manager resolve peer dependencies. It has been addressed in v7 and upgrading should be very straight-forward.

There's still one relevant part to my issue which is the fact that the docs  are currently incorrect. There's no mention of needing to add @pixi/assets on npm and I only found that package by looking through the repo. The docs assume it can be imported from the pixi.js package, which is wrong

**Comment by SuperSodaSea** [comment]
By the way, I think we need to add links for different version of API Docs (latest, v7.x, v6.x, dev, etc.) on the navbar of [pixijs.com](https://pixijs.com) instead of just the latest release version, in order to avoid confusion.

**Comment by liana-p** [comment]
> By the way, I think we need to add links for different version of API Docs (latest, v7.x, v6.x, dev, etc.) on the navbar of [pixijs.com](https://pixijs.com) instead of just the latest release version, in order to avoid confusion.

Yes, I was wondering if that's a thing. At first I thought I was on the wrong docs, I noticed there's a release doc and a dev doc, but they both had the same content for the assets page

**Comment by SuperSodaSea** [comment]
[https://pixijs.download/v6.x/docs/PIXI.Assets.html](https://pixijs.download/v6.x/docs/PIXI.Assets.html)
The updated document is here! Hope you enjoy it. :smiley:
I'll add links for different versions of API Docs for [pixijs.com](https://pixijs.com) in the near future.

**Comment by F8ER** [comment]
Dear Developers,

Thank you for the awesome Project!

I am sorry, but apparently, there is some interesting issue when trying the following:

```
ERROR in ./node_modules/@pixi/assets/lib/utils/checkExtension.mjs 5:14-32
export 'utils' (imported as 'utils') was not found in '@pixi/core' (possible exports: AbstractBatchRenderer, AbstractMultiResource, AbstractRenderer, ArrayResource, Attribute, BaseImageResource, BaseRenderTexture, BaseTexture, BatchDrawCall, BatchGeometry, BatchPluginFactory, BatchRenderer, BatchShaderGenerator, BatchSystem, BatchTextureArray, Buffer, BufferResource, CanvasResource, ContextSystem, CubeResource, ExtensionType, Filter, FilterState, FilterSystem, Framebuffer, FramebufferSystem, GLFramebuffer, GLProgram, GLTexture, Geometry, GeometrySystem, IGLUniformData, INSTALLED, ImageBitmapResource, ImageResource, MaskData, MaskSystem, ObjectRenderer, Program, ProjectionSystem, Quad, QuadUv, RenderTexture, RenderTexturePool, RenderTextureSystem, Renderer, Resource, SVGResource, ScissorSystem, Shader, ShaderSystem, SpriteMaskFilter, State, StateSystem, StencilSystem, System, Texture, TextureGCSystem, TextureMatrix, TextureSystem, TextureUvs, UniformGroup, VERSION, VideoResource, ViewableBuffer, autoDetectRenderer, autoDetectResource, checkMaxIfStatementsInShader, createUBOElements, defaultFilterVertex, defaultVertex, extensions, generateProgram, generateUniformBufferSync, getTestContext, getUBOData, resources, systems, uniformParsers)
```

The code:

```
import * as PIXI from 'pixi.js';
import * as PixiParticles from '@pixi/particle-emitter';
import { Assets } from '@pixi/assets';

// ...

class SomeClass
{
    // ...

    async someMethod()
    {
        // ...
        await _loadTextures()
            .then // ...
        // ...
    }

    async _loadTextures()
    {
        const assetsRaw = Object.keys(imports.textures).reduce((result, key) => {
            // eslint-disable-next-line no-param-reassign
            result = {
                urls: imports.textures[key],
                alias: key
            };

            return result;
        }, {});

        console.log('Assets raw', assetsRaw);

        return Assets.load(assetsRaw);
    }

    // ...
}
```

What might it be?

Best and kind regards

**Comment by SuperSodaSea** [comment]
@F8ER You are mixing different versions of `@pixi/assets` (7.0.5) and `pixi.js` (6.5.8). You need to upgrade `pixi.js` to 7.0.5.

**Comment by F8ER** [comment]
> @F8ER You are mixing different versions of `@pixi/assets` (7.0.5) and `pixi.js` (6.5.8). You need to upgrade `pixi.js` to 7.0.5, or downgrade `@pixi/assets` to 6.5.8.

Please accept my sincere appreciation! May I ask where could one find the version dependencies stated if any?

**Comment by SuperSodaSea** [comment]
@F8ER Every time we release new versions for all packages in <https://github.com/pixijs/pixijs>, so just keep them the same version. For other packages like `@pixi/particle-emitter`, you can check their `package.json` file.

**Comment by F8ER** [comment]
> @F8ER Every time we release new versions for all packages in https://github.com/pixijs/pixijs, so just keep them the same version. For other packages like `@pixi/particle-emitter`, you can check their `package.json` file.

I am sorry, but the project in question has some legacy stuff and there's quite out in there... so... it's currently required to preserve the version stated above, hence the clarification instead an upgrade.
